conftest.py

```diff
@@ -232,18 +232,24 @@ def extra_checks_test_wrapper(browser, trace_hiwire_refs, trace_pyproxies):
     if trace_pyproxies:
         browser.enable_pyproxy_tracing()
         init_num_proxies = browser.get_num_proxies()
-    a = yield
+    err = None
     try:
-        # If these guys cause a crash because the test really screwed things up,
-        # we override the error message with the better message returned by
-        # a.result() in the finally block.
-        browser.disable_pyproxy_tracing()
-        browser.restore_state()
+        result = yield
+    except Exception as e:
+        err = e
     finally:
-        # if there was an error in the body of the test, flush it out by calling
-        # get_result (we don't want to override the error message by raising a
-        # different error here.)
-        a.get_result()
+        try:
+            # If these guys cause a crash because the test really screwed things up,
+            # we override the error message with the better message returned by
+            # a.result() in the finally block.
+            browser.disable_pyproxy_tracing()
+            browser.restore_state()
+        finally:
+            try:
+                if err:
+                    raise err
+            finally:
+                del err
```

Comment:
I guess this would be an improvement here:
```suggestion
        # If the test really screwed things up, these cleanup functions might fail.
        # Suppress any errors in this case so we keep the orignal error
        cm = contextlib.suppress(Exception) if err else contextlib.nullcontext()
        with cm:
            browser.disable_pyproxy_tracing()
            browser.restore_state()
```